pandas-stubs/core/indexes/base.pyi

```diff
@@ -902,18 +910,166 @@ class Index(IndexOpsMixin[S1], ElementOpsMixin[S1]):
     def __rmul__(
         self: Index[T_COMPLEX], other: np_ndarray_complex | Index[complex]
     ) -> Index[complex]: ...
+    @overload
+    def __truediv__(
+        self: Index[Never], other: complex | ArrayLike | SequenceNotStr[S1] | Index
+    ) -> Index: ...
+    @overload
+    def __truediv__(self, other: Index[Never]) -> Index: ...
+    @overload
+    def __truediv__(self: Index[bool], other: np_ndarray_bool) -> Never: ...
+    @overload
+    def __truediv__(
+        self: Index[Timedelta],
+        other: np_ndarray_bool | np_ndarray_complex | np_ndarray_dt,
+    ) -> Never: ...
+    @overload
+    def __truediv__(
+        self: Supports_ProtoTrueDiv[_T_contra, S2],
+        other: _T_contra | Sequence[_T_contra],
+    ) -> Index[S2]: ...
+    @overload
+    def __truediv__(
+        self: Index[int],
+        other: np_ndarray_bool | Index[bool],
+    ) -> Index[float]: ...
+    @overload
+    def __truediv__(
+        self: Index[bool] | Index[int],
+        other: Just[int] | Sequence[int] | np_ndarray_anyint | Index[int],
+    ) -> Index[float]: ...
+    @overload
+    def __truediv__(
+        self: Index[float],
+        other: np_ndarray_bool | np_ndarray_anyint | Index[bool] | Index[int],
+    ) -> Index[float]: ...
+    @overload
+    def __truediv__(
+        self: Index[complex],
+        other: np_ndarray_bool | np_ndarray_anyint | Index[bool] | Index[int],
+    ) -> Index[complex]: ...
+    @overload
+    def __truediv__(
+        self: Index[bool] | Index[int],
+        other: Just[float] | Sequence[Just[float]] | np_ndarray_float | Index[float],
+    ) -> Index[float]: ...
+    @overload
+    def __truediv__(
+        self: Index[T_COMPLEX],
+        other: Just[float] | Sequence[Just[float]] | np_ndarray_float | Index[float],
+    ) -> Index[T_COMPLEX]: ...
+    @overload
+    def __truediv__(
+        self: Index[T_COMPLEX],
+        other: (
+            Just[complex]
+            | Sequence[Just[complex]]
+            | np_ndarray_complex
+            | Index[complex]
+        ),
+    ) -> Index[complex]: ...
+    @overload
+    def __truediv__(
+        self: Index[Timedelta],
```

Comment:
Really would prefer that we don't support operations on `Index[Timedelta]` because that is a `TimedeltaIndex`, so if somehow our stubs are producing `Index[Timedelta]`, that should be fixed.